### PR TITLE
imgur-album-downloader (new formula)

### PR DESCRIPTION
**Pull request by sgsunder**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Adds formula for [imgur album downloader](https://github.com/alexgisby/imgur-album-downloader/commits/master)

### CODE REVIEW

**Comment by ilovezfs** [comment]
@sgsunder Something's off with the test's syntax:
```
Testing imgur-album-downloader
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20160507-69170-aw5osl.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Taps/homebrew/homebrew-core/Formula/imgur-album-downloader.rb --verbose
==> /usr/local/Cellar/imgur-album-downloader/1/bin/imgur-album-downloader --help
Error: URL must be a valid Imgur Album

How to use
=============

Quickly and easily download an album from Imgur.

Format:
    $ python imguralbum.py [album URL] [destination folder]

Example:
    $ python imguralbum.py http://imgur.com/a/uOOju#6 /Users/alex/images

If you omit the dest folder name, the utility will create one with the same name
as the album
(for example for http://imgur.com/a/uOOju it'll create uOOju/ in the cwd)

Error: imgur-album-downloader: failed
Failed executing: /usr/local/Cellar/imgur-album-downloader/1/bin/imgur-album-downloader --help
```

**Comment by sgsunder** [comment]
Fixed the test. Wasn't supposed to use a '--help' flag.

File: Formula/imgur-album-downloader.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class ImgurAlbumDownloader < Formula
+  desc "Command Line Utility to Download Imgur Albums"
+  homepage "https://github.com/sgsunder/imgur-album-downloader"
+  url "https://github.com/alexgisby/imgur-album-downloader.git", :revision => "dbaee1e342c1b46c022fccd26c294e251ac04015"
```
Comment (bfontaine) [inline Formula/imgur-album-downloader.rb]: We need a tagged release here.

File: Formula/imgur-album-downloader.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  version "1"
+
+  bottle :unneeded
+
+  depends_on "python3"
+
+  def install
+    mv "imguralbum.py", "imgur-album-downloader" # Rename Command Line Script
+    bin.install "imgur-album-downloader"
```
Comment (bfontaine) [inline Formula/imgur-album-downloader.rb]: You can use `bin.install "imguralbum.py" => "imgur-album-downloader"`.

File: Formula/imgur-album-downloader.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  test do
+    system "#{bin}/imgur-album-downloader" # This should print the help message and return with exit code 0
```
Comment (bfontaine) [inline Formula/imgur-album-downloader.rb]: Can’t we do something more substantial that just printing the help? E.g. download a small album?

File: Formula/imgur-album-downloader.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class ImgurAlbumDownloader < Formula
+  desc "Command Line Utility to Download Imgur Albums"
+  homepage "https://github.com/sgsunder/imgur-album-downloader"
```
Comment (bfontaine) [inline Formula/imgur-album-downloader.rb]: This gives a 404; should probably be `https://github.com/alexgisby/imgur-album-downloader` instead.

**Comment by sgsunder** [comment]
I'm not sure what you mean by "tagged release." Is it not acceptable to fix to a specific commit?

File: Formula/imgur-album-downloader.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/imgur-album-downloader.rb]: I wonder if one of the preexisting test fixtures could be uploaded and then re-downloaded here

**Comment by DomT4** [comment]
I'm afraid not. We require a stable, tagged upstream release for formulae in this repository.

**Comment by DomT4** [comment]
A tagged release in this context being something similar to [as seen here](https://github.com/Kentzo/git-archive-all/releases).

**Comment by ilovezfs** [comment]
@sgsunder you need to file an issue with upstream (https://github.com/alexgisby/imgur-album-downloader/issues) and ask them to create a tag.

Here are some examples of such a request:
https://github.com/scantailor/scantailor/issues/204
https://github.com/neurolabs/henplus/issues/41
https://github.com/att/ast/issues/3

**Comment by sgsunder** [comment]
Will do. I'll keep this thread posted for a response.

File: Formula/imgur-album-downloader.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://github.com/alexgisby/imgur-album-downloader"
+  url "https://github.com/alexgisby/imgur-album-downloader.git", :revision => "dbaee1e342c1b46c022fccd26c294e251ac04015"
+  version "1"
+
+  bottle :unneeded
+
+  depends_on "python3"
+
+  def install
+    bin.install "imguralbum.py" => "imgur-album-downloader" # Rename Command Line Script
```
Comment (mikemcquaid) [inline Formula/imgur-album-downloader.rb]: Can remove the comment.

File: Formula/imgur-album-downloader.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    system "#{bin}/imgur-album-downloader", "http://imgur.com/a/ej9g4" # Download test album
```
Comment (mikemcquaid) [inline Formula/imgur-album-downloader.rb]: Can remove the comment.

**Comment by DomT4** [comment]
It looks like there's no prospect of a stable release here from a stable repo any time soon; we've been waiting since early May & still no progress.

Will happily re-review this PR or a fresh one if a stable release is ever published, but for now I'm going to close with the presumption there's not one coming soon. Thank you nonetheless for your submission to Homebrew @sgsunder ❤️.

**Pull request closed without merging**